src/Query/Expression.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ClickHouseDB\Query;
+
+/**
+ * Pass expression "as is" to be sent and executed at server.
+ * P.ex.: `new Expression("UUIDStringToNum('0f372656-6a5b-4727-a4c4-f6357775d926')");`
+ */
+class Expression
+{
+    /** @var string */
+    private $expression;
```

Comment:
There was different intention and reasoning for `Type` and `Expression`. 
`Expression` was made to bypass quoting. Inspired by other DB-related libs. But, btw, not necessarily in a best way implement this intention. I just found it less obtrusive and easier to merge in future back to my fork.
`Type`, as I can see, introduced for different reason - to allow safely and in a controlled fashion add new, potentially complex types(what about `Date` type, f.ex.?).

While `Expression` seems to be universal here I'm against replacing `Type` with `Expression`, because `Expression` is a last resort. Unsafe one, I must say. Should be used only when there are no other options. In theory, with time, `Expression` could be replaced with some support for functions from library itself. Like `new Function\UUIDStringToNum($uuid)` instead of `new Expression("UUIDStringToNum('0f372656-6a5b-4727-a4c4-f6357775d926')")`.

Reply:
kk, let's just leave it like it is

Reply:
So, maybe, `Expression` could become either a higher level (abstract?) class or even Interface and `Type` can be a descendant to it. Because you're right in one aspect - `Expression` could solve a wider range of requirements including ones that are solved by `Type`. 
But it will require to refactor this part in `src/Quote/StrictQuoteLine.php`:
```php
            $value = ValueFormatter::formatValue($value, false);

            if (is_float($value) || is_int($value)) {
                return (string) $value;
            }

            if (is_string($value) && $encode) {
```
if you want `Expression::getValue()` to return string. Because otherwise string 
`"UUIDStringToNum('0f372656-6a5b-4727-a4c4-f6357775d926')"`
will become 
`'\'UUIDStringToNum(\\\'0f372656-6a5b-4727-a4c4-f6357775d926\\\')\''`
after quoting.

Reply:
You're right it might need some tweak in that part of code. But see this line

https://github.com/smi2/phpClickHouse/blob/master/src/Quote/StrictQuoteLine.php#L74

If replaced by `if ($value instanceof Expression) {` it might do the trick.

BTW I'd definitely be for `interface Expression` rather than `abstract class Expression`

Reply:
Yes, missed that one. So it should be something like this IMO:
```php
            if ($value instanceof NumericType) {
                $encode = $value->doQuoting();
            }
```

But in this case abstract class AND interface required. Abstract is required to define default `doQuoting() : boolean`. I think I can do it after this one is merged. Because, obviously, it will be refactoring of `Type` and `Expression`.

Reply:
Yes. I'd name it something like `needsEncoding() : bool` and abstract class is IMO not required, we don't really need default value so interface will do. But for sure in next PR :)